sourcegraph/cody
# Fix cursor position after submitting mention items

Based on https://github.com/sourcegraph/cody/pull/4777
Fixes https://linear.app/sourcegraph/issue/SRCH-672/cursor-jumps-around-with-mentions

I think the original issue that was reported [here](https://linear.app/sourcegraph/issue/SRCH-672/cursor-jumps-around-with-mentions) has already been partially fixed, but there is still a problem with two-step provider mentions, I think, and this PR tries to solve this. 

So the problem was that when you have this text `Hello World` and you're trying to add remote file mentions like `Hello @<cursor> World` and select some repository from the suggestion, the result will be `Hello @<cursor>github.com/sourcegraph/sourcegraph: World`, you can see the problem that the cursor stays in the same position, and this breaks two steps provider mentions flow. 

Another problem is that we rely too much on the `@` symbol when we're trying to find a current mention in progress to insert text from the suggestion panel, for example when you have `Hello@ World@` and you're typing `Hello@ @<cursor> World@`, the `@` symbol that actually triggered mention panel won't be used but I think only first `@` after the word `Hello` will be used and as a result, we will have `Hello<mention> @ World@`, which is not correct I guess. 

@keegancsmith I've seen that you recently added some logic about similar prefixes [here](https://github.com/sourcegraph/cody/pull/4654), I'm not sure I fully understand this, I tried to fix only two steps mentions in this PR, can you test this manually since you obviously have more cases to check this properly

## Test plan
- Build vscode extension and check that mentions works properly (different types files, symbols, repository, remote files, remote repositories)
- Run Cody Web demo (`cd ./web` and `pnpm dev`) and check that all mentions work properly and you see no jumps as you select some mentions with keyboard or with mouse.
